3.x: Fix code and docs regarding driver name

dkropachev opened the pull request:
Fixes driver name in comments, logs, docs.
Fixes test to check driver name and version on server side.

dkropachev committed:
Fix driver name in comments and docs

dkropachev committed:
Add scylla.version property

It doesn't do anything, just to keep in organized

dkropachev committed:
Log correct dirver name

dkropachev committed:
Check if driver reports correct driver name to the server

dkropachev commented:
To be rebased after https://github.com/scylladb/java-driver/pull/407 to cicd become green.

Bouncheck commented on pom.xml:
```diff
@@ -50,6 +50,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <cassandra.version>3.11.2</cassandra.version>
+        <scylla.version/>
```
I don't understand the purpose of this property.

dkropachev replied:
We use it to set scylla version, this change is only to put it into the pom to make it visible there.

Bouncheck reviewed (approved):
LGTM

dkropachev committed:
Install peotry with --no-root

Otherwise it fails with:
```
Installing the current project: java-driver (3.7.1)

Error: The current project could not be installed: No file/folder found for package java-driver
If you do not want to install the current project use --no-root.
If you want to use Poetry only for dependency management but not for packaging, you can disable package mode by setting package-mode = false in your pyproject.toml file.
If you did intend to install the current project, you may need to set `packages` in your pyproject.toml file.
```
